homeassistant/components/nobo_hub/climate.py

```diff
@@ -0,0 +1,208 @@
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(self, zone_id, hub: nobo, override_type):
+        """Initialize the climate device."""
+        self._id = zone_id
+        self._nobo = hub
+        self._attr_unique_id = hub.hub_serial + ":" + zone_id
+        self._attr_name = hub.zones[self._id][ATTR_NAME]
+        self._attr_hvac_mode = HVACMode.AUTO
+        self._attr_hvac_modes = [HVACMode.HEAT, HVACMode.AUTO]
+        self._override_type = override_type
+
+    async def async_added_to_hass(self) -> None:
+        """Register callback from hub."""
+        self._nobo.register_callback(self._after_update)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Deregister callback from hub."""
+        self._nobo.deregister_callback(self._after_update)
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new target HVAC mode, if it's supported."""
+        if hvac_mode not in self.hvac_modes:
+            _LOGGER.warning(
+                "Zone %s '%s' called with unsupported HVAC mode '%s'",
+                self._id,
+                self._attr_name,
+                hvac_mode,
+            )
+            return
+        if hvac_mode == HVACMode.AUTO:
+            await self.async_set_preset_mode(PRESET_NONE)
+        elif hvac_mode == HVACMode.HEAT:
+            await self.async_set_preset_mode(PRESET_COMFORT)
+        self._attr_hvac_mode = hvac_mode
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new zone override."""
+        if self._nobo.zones[self._id][ATTR_OVERRIDE_ALLOWED] != "1":
+            return
+        if preset_mode == PRESET_ECO:
+            mode = nobo.API.OVERRIDE_MODE_ECO
+        elif preset_mode == PRESET_AWAY:
+            mode = nobo.API.OVERRIDE_MODE_AWAY
+        elif preset_mode == PRESET_COMFORT:
+            mode = nobo.API.OVERRIDE_MODE_COMFORT
+        else:  # PRESET_NONE
+            mode = nobo.API.OVERRIDE_MODE_NORMAL
+        await self._nobo.async_create_override(
+            mode,
+            self._override_type,
+            nobo.API.OVERRIDE_TARGET_ZONE,
+            self._id,
+        )
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        low, high = None, None
+        if ATTR_TARGET_TEMP_LOW in kwargs:
+            low = int(kwargs.get(ATTR_TARGET_TEMP_LOW))  # type: ignore[arg-type]  # ATTR_TARGET_TEMP_LOW is float if set
+        if ATTR_TARGET_TEMP_HIGH in kwargs:
+            high = int(kwargs.get(ATTR_TARGET_TEMP_HIGH))  # type: ignore[arg-type]  # ATTR_TARGET_TEMP_HIGH is float if set
+        if low is not None:
+            if high is not None:
+                low = min(low, high)
+            elif self.target_temperature_high is not None:
+                low = min(low, int(self.target_temperature_high))
+        elif high is not None and self.target_temperature_low is not None:
```

Comment:
high can't be None of low is not None. So we should probably indent this case within the above block.

Reply:
This is for the case if only `ATTR_TARGET_TEMP_HIGH` is passed in `kwargs`.

Can assume that either both or none of `ATTR_TARGET_TEMP_LOW` and `ATTR_TARGET_TEMP_HIGH` are set in `kwargs`?

Can we also assume that low <= high if they are set?

And even better, can we also assume that not having `TARGET_TEMPERATURE` is respected?

If so, we can simplify to:
```
        await self._nobo.async_update_zone(
            self._id,
            temp_comfort_c=round(kwargs[ATTR_TARGET_TEMP_HIGH]),
            temp_eco_c=round(kwargs[ATTR_TARGET_TEMP_LOW])
        )
```

I tried to look into the climate component code based on an earlier comment, but I didn't find any assertions of the above assumptions.

Reply:
We can assume that both high and low target temps are passed in the service call if any of them is passed. The service schema validates this.

Reply:
Thanks! Simplified using this assumption.